pkg/observable/channel/replay_test.go

```diff
@@ -300,3 +301,78 @@ func TestReplayObservable_Last_Blocks_And_Times_Out(t *testing.T) {
 	require.ElementsMatch(t, []int{5, 4, 3, 2}, replayObsvbl.Last(ctx, 4))
 	require.ElementsMatch(t, []int{5, 4, 3, 2, 1}, replayObsvbl.Last(ctx, 5))
 }
+
+func TestReplayObservable_SubscribeFromLatestBufferedOffset(t *testing.T) {
+	receiveTimeout := 100 * time.Millisecond
+	values := []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
+
+	tests := []struct {
+		name             string
+		replayBufferSize int
+		endOffset        int
+		expectedValues   []int
+	}{
+		{
+			name:             "endOffset = replayBufferSize",
+			replayBufferSize: 8,
+			endOffset:        8,
+			expectedValues:   values[2:], // []int{2, 3, 4, 5, ...},
+		},
+		{
+			name:             "endOffset < replayBufferSize",
+			replayBufferSize: 10,
+			endOffset:        2,
+			expectedValues:   values[8:], // []int{8, 9},
+		},
+		{
+			name:             "endOffset > replayBufferSize",
+			replayBufferSize: 8,
+			endOffset:        10,
+			expectedValues:   values[2:],
+		},
+		{
+			name:             "replayBufferSize < eldOffset < numBufferedValues ",
+			replayBufferSize: 20,
+			endOffset:        15,
```

Comment:
1. WHat's eldOffset?
2. Assuming (1) is a typo of end, the `<` is still incorrect